go/proxyd/backend.go

```diff
@@ -395,13 +423,16 @@ type BackendGroup struct {
 	Backends []*Backend
 }
 
-func (b *BackendGroup) Forward(ctx context.Context, rpcReq *RPCReq) (*RPCRes, error) {
+func (b *BackendGroup) Forward(ctx context.Context, rpcReq *RPCReq) (*RPCRes, uint64, error) {
 	rpcRequestsTotal.Inc()
 
 	for _, back := range b.Backends {
+		// The blockNum must precede the forwarded RPC to establish a synchronization point
+		blockNum := back.BlockNumber()
```

Comment:
I'm wary of hitting the backend with a block number call on every RPC. This will effectively double the number of requests we're making to the upstream, and introduces a race condition between getting the block number and getting the actual RPC result. I'd much rather not cache requests that have the `latest` block tag. This will eliminate this complexity. Our cache hit rate will suffer a bit, but I'm ok paying a performance penalty in favor of result consistency.

Reply:
I agree that we should just never cache anything that has the block tag `latest` or `pending`

Reply:
We can always update the logic in the future after shipping something more simple if our miss rate is too high